src/main/java/org/terasology/eventualSkills/systems/EventualSkillsClientSystem.java

```diff
@@ -1,51 +1,105 @@
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
 package org.terasology.eventualSkills.systems;
 
+import org.terasology.engine.SimpleUri;
 import org.terasology.entitySystem.entity.EntityRef;
 import org.terasology.entitySystem.event.ReceiveEvent;
 import org.terasology.entitySystem.systems.BaseComponentSystem;
 import org.terasology.entitySystem.systems.RegisterMode;
 import org.terasology.entitySystem.systems.RegisterSystem;
 import org.terasology.eventualSkills.EventualSkillsButton;
+import org.terasology.eventualSkills.components.EntitySkillsComponent;
 import org.terasology.eventualSkills.ui.EventualSkillsTraining;
 import org.terasology.input.ButtonState;
+import org.terasology.input.Input;
+import org.terasology.input.InputSystem;
 import org.terasology.logic.players.LocalPlayer;
+import org.terasology.logic.players.event.LocalPlayerInitializedEvent;
 import org.terasology.network.ClientComponent;
+import org.terasology.notifications.events.ShowNotificationEvent;
+import org.terasology.notifications.model.Notification;
+import org.terasology.notifications.events.ExpireNotificationEvent;
+import org.terasology.nui.Color;
+import org.terasology.nui.FontColor;
 import org.terasology.registry.In;
 import org.terasology.rendering.nui.NUIManager;
+import org.terasology.unicode.EnclosedAlphanumerics;
 
 @RegisterSystem(RegisterMode.CLIENT)
 public class EventualSkillsClientSystem extends BaseComponentSystem {
+
+    private static final String NOTIFICATION_ID = "EventualSkills:noSkills";
+
     @In
     LocalPlayer localPlayer;
     @In
     NUIManager nuiManager;
+    @In
+    InputSystem inputSystem;
 
     @Override
     public void initialise() {
         super.initialise();
         nuiManager.getHUD().addHUDElement("EventualSkills:NoSkillsNagWidget");
     }
 
+    /**
+     * Get a formatted representation of the primary {@link Input} associated with the given button binding.
+     * <p>
+     * If the display name of the primary bound key is a single character this representation will be the encircled
+     * character. Otherwise the full display name is used. The bound key will be printed in yellow.
+     * <p>
+     * If no key binding was found the text "n/a" in red color is returned.
+     *
+     * @param button the URI of a bindable button
+     * @return a formatted text to be used as representation for the player
+     */
+    //TODO: put this in a common place? Duplicated in Dialogs and InGameHelp
```

Comment:
Do we have an issue for that yet?

Reply:
no, as I don't know where exactly to file that...